src/mirrorNode.ts

```diff
@@ -0,0 +1,84 @@
+import axios, { AxiosResponse } from 'axios';
+import { MessageObject } from './types/MessageObject';
+import { NFT } from './types/NFT';
+
+export type NetworkType =
+  'hedera_test' | 'hedera_main' | 'lworks_test' | 'lworks_main'| 'arkhia_test' | 'arkhia_main';
+
+export const TLD_TOPIC_ID = '0.0.47954429';
+
+export enum NetworkBaseURL {
+  'hedera_test' = 'https://testnet.mirrornode.hedera.com/api/v1',
+  'hedera_main' = 'https://mainnet-public.mirrornode.hedera.com/api/v1',
+  'lworks_test' = 'https://testnet.mirror.lworks.io/api/v1',
+  'lworks_main' = 'https://mainnet.mirror.lworks.io/api/v1',
+  'arkhia_test' = 'https://hedera.testnet.arkhia.io/api/v1',
+  'arkhia_main' = 'https://hedera.mainnet.arkhia.io/api/v1'
+}
+
+// Max page size allowed by hedera nodes
+const MAX_PAGE_SIZE = 100;
+
+export class MirrorNode {
+  networkType: NetworkType;
+  baseUrl: string;
+  authKey: string;
+
+  constructor(networkType: NetworkType, authKey = '') {
+    this.networkType = networkType;
+    this.baseUrl = this.getBaseUrl();
+    this.authKey = authKey;
+  }
+
+  async getTopicMessages(topicId: string, offset: number): Promise<MessageObject[]> {
+    let sequenceNumber: number;
+    offset === 0 ? sequenceNumber = offset + 1 : sequenceNumber = (offset + 1) * MAX_PAGE_SIZE;
+
+    const url =
+    `${this.getBaseUrl()}/topics/${topicId}/messages/?sequenceNumber=gte:${sequenceNumber}&limit=${MAX_PAGE_SIZE}`;
```

Comment:
we are setting base url in contructor. We can call this.baseUrl instead I believe